source/common/http/filter/fault_filter.cc

```diff
@@ -142,6 +150,21 @@ void FaultFilter::abortWithHTTPStatus() {
   callbacks_->requestInfo().setResponseFlag(Http::AccessLog::ResponseFlag::FaultInjected);
 }
 
+bool FaultFilter::matchesTargetCluster(HeaderMap& headers) {
+  bool matches = true;
+
+  if (!config_->targetCluster().empty()) {
+    // TODO PERF Eliminate repeated route matching
+    const Router::Route* route = callbacks_->routeTable().route(headers);
+
+    matches &= ((nullptr != route) && (nullptr != route->routeEntry()));
```

Comment:
i'd just have all of these in

matches = route && route->routeEntry() && route->routeEntry()->clusterName() == config_->targetCluster;